easybuild/easyconfigs/p/PyTorch/PyTorch-2.1.0-foss-2023a.eb

```diff
@@ -0,0 +1,157 @@
+name = 'PyTorch'
+version = '2.1.0'
+
+homepage = 'https://pytorch.org/'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
+PyTorch is a deep learning framework that puts Python first."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+source_urls = [GITHUB_RELEASE]
+sources = ['%(namelower)s-v%(version)s.tar.gz']
+patches = [
+    'PyTorch-1.7.0_disable-dev-shm-test.patch',
+    'PyTorch-1.11.1_skip-test_init_from_local_shards.patch',
+    'PyTorch-1.12.1_add-hypothesis-suppression.patch',
+    'PyTorch-1.12.1_fix-test_cpp_extensions_jit.patch',
+    'PyTorch-1.12.1_fix-TestTorch.test_to.patch',
+    'PyTorch-1.12.1_skip-test_round_robin.patch',
+    'PyTorch-1.13.1_fix-gcc-12-warning-in-fbgemm.patch',
+    'PyTorch-1.13.1_fix-protobuf-dependency.patch',
+    'PyTorch-1.13.1_fix-warning-in-test-cpp-api.patch',
+    'PyTorch-1.13.1_skip-failing-singular-grad-test.patch',
+    'PyTorch-1.13.1_skip-tests-without-fbgemm.patch',
+    'PyTorch-2.0.1_avoid-test_quantization-failures.patch',
+    'PyTorch-2.0.1_fix-skip-decorators.patch',
+    'PyTorch-2.0.1_fix-ub-in-inductor-codegen.patch',
+    'PyTorch-2.0.1_fix-vsx-loadu.patch',
+    'PyTorch-2.0.1_no-cuda-stubs-rpath.patch',
+    'PyTorch-2.0.1_skip-failing-gradtest.patch',
+    'PyTorch-2.0.1_skip-test_shuffle_reproducibility.patch',
+    'PyTorch-2.0.1_skip-tests-skipped-in-subprocess.patch',
+    'PyTorch-2.0.1_workaround-gcc12-avx512-optimizer-bug.patch',
```

Comment:
This patch can be removed after #19180 fixed the root cause.

Reply:
Thank you very much.  Does it do any harm to leave it in?

Some places will have installed GCCcore/12.3.0 already, and unless they recompile they will be hit by this.

Reply:
Yes, I see harm in 2 ways:
1) It degrades performance by doing additional work on affected configurations
2) Failing here makes the Compiler-Bug visible causing people to be aware of it. Masking it with this patch will make people continue using the unpatched compiler resulting in potentially faulty programs silently producing wrong results.

Reply:
OK, thank you!

I will recompile GCCcore for myself, and once I have verified that it works I'll update this PR.

Then I'll ask for GCCcore to be recompiled on the parts of our cluster supporting AVX512BW/VL - I guess we will have to prevent new jobs from starting while the GCCcore module is briefly gone :-)

Reply:
Patch removed.